src/layout/core/tree.rs

```diff
@@ -99,6 +116,28 @@ impl LayoutTree {
         self.active_container = None;
     }
 
+    /// Gets the root container of the active container.
+    ///
+    /// If there is no active container, searches the path.
+    pub fn root_container_ix(&self) -> Option<NodeIndex> {
```

Comment:
Rework this function to work given an NodeIndex that is not an Output or Root node!